pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -4438,6 +4438,66 @@ status:
 	}
 }
 
+func TestStopSidecars_WithInjectedSidecarsNoTaskSpecSidecars(t *testing.T) {
+	taskRun := parse.MustParseV1beta1TaskRun(t, `
+metadata:
+  name: test-taskrun-injected-sidecars
+  namespace: foo
+spec:
+  taskRef:
+    name: test-task
+status:
+  podName: test-taskrun-injected-sidecars-pod
+  conditions:
+  - message: Build succeeded
+    reason: Build succeeded
+    status: "True"
+    type: Succeeded
+  sidecars:
+  - container: injected-sidecar
+    imageID: image-id
+    name: sidecar
+    running:
+      startedAt: "2022-01-01T00:00:00Z"
+`)
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-taskrun-injected-sidecars-pod",
+			Namespace: "foo",
+		},
+	}
+
+	d := test.Data{
+		Pods:     []*corev1.Pod{pod},
+		TaskRuns: []*v1beta1.TaskRun{taskRun},
+		Tasks:    []*v1beta1.Task{simpleTask},
+	}
+
+	testAssets, cancel := getTaskRunController(t, d)
+	defer cancel()
+	c := testAssets.Controller
+	clients := testAssets.Clients
+	reconcileErr := c.Reconciler.Reconcile(testAssets.Ctx, getRunName(taskRun))
+
+	// We do not expect an error
+	if reconcileErr != nil {
+		t.Errorf("Expected no error to be returned by reconciler: %v", reconcileErr)
+	}
+
+	// Verify that the pod was retrieved.
+	getPodFound := false
+	for _, action := range clients.Kube.Actions() {
+		if action.Matches("get", "pods") {
+			getPodFound = true
+			break
+		}
```

Comment:
it might be simpler to use `clients.Kube.CoreV1().Pods("foo").Get("test-taskrun-injected-sidecars-pod")` (not sure of exact syntax) since this relies less on implementation details and is just a bit more readable

Reply:
I took that implementation from the test above for consistency, but it is checking that the reconciler actually reached out to get the actual pod which means that it made it to the point where it had sidecars to stop

Reply:
I think our existing tests could probably use some improvement here; they seem to test implementation details rather than functionality visible to users. I would create a test that has a pod with completed step containers and running injected sidecar containers, and tests that the injected containers are replaced with the nop image.